criu/util.c

```diff
@@ -709,26 +716,32 @@ int is_root_user(void)
 	return 1;
 }
 
+/* is_empty_dir will always close the FD dirfd */
 int is_empty_dir(int dirfd)
 {
 	int ret = 0;
 	DIR *fdir = NULL;
 	struct dirent *de;
 
 	fdir = fdopendir(dirfd);
-	if (!fdir)
-		return -1;
+	if (!fdir) {
+		ret = -1;
+		goto out;
+	}
 
 	while ((de = readdir(fdir))) {
 		if (dir_dots(de))
 			continue;
 
-		goto out;
+		goto out_closedir;
 	}
 
 	ret = 1;
-out:
+
+out_closedir:
 	closedir(fdir);
+out:
+	close_safe(&dirfd);
```

Comment:
Now you have double close. What I was thinking of is:

```
+/*
+ * is_empty_dir will always close the FD dirfd: either implicitly
+ * via closedir or explicitly in case fdopendir had failed
+ */
 int is_empty_dir(int dirfd)
 {
        int ret = 0;
@@ -716,8 +727,10 @@ int is_empty_dir(int dirfd)
        struct dirent *de;
 
        fdir = fdopendir(dirfd);
-       if (!fdir)
+       if (!fdir) {
+               close_safe(&dirfd);
                return -1;
+       }
 
        while ((de = readdir(fdir))) {
                if (dir_dots(de))
```

Reply:
Thanks. Changed and pushed.